tests/baselines/reference/strictFunctionTypes1.types

```diff
@@ -53,16 +53,16 @@ const x1 = f1(fo, fs);  // (x: string) => void
 >fs : (x: string) => void
 
 const x2 = f2("abc", fo, fs);  // "abc"
->x2 : string
->f2("abc", fo, fs) : string
+>x2 : "abc"
+>f2("abc", fo, fs) : "abc"
 >f2 : <T>(obj: T, f1: (x: T) => void, f2: (x: T) => void) => T
 >"abc" : "abc"
 >fo : (x: Object) => void
 >fs : (x: string) => void
 
 const x3 = f3("abc", fo, fx);  // "abc" | "def"
->x3 : Object
->f3("abc", fo, fx) : Object
+>x3 : "def" | "abc"

```

Comment:
We should discuss this later, but this inference seems bad

Reply:
No, this inference is correct. The `fx` argument is a function taking a function, and since contra-contra-variant is the same as co-variant, we make two co-variant inferences, `"abc"` and `"def"`. Per normal rules we union together literal inferences for the same type parameter, so we end up with `"abc" | "def"`.